pkg/controllers/provisioning/scheduling/topologynodefilter.go

```diff
@@ -27,102 +27,60 @@ import (
 // included for topology counting purposes. This is only used with topology spread constraints as affinities/anti-affinities
 // always count across all nodes. A nil or zero-value TopologyNodeFilter behaves well and the filter returns true for
 // all nodes.
-type TopologyNodeFilter struct {
-	nodeSelector      map[string]string
-	nodeSelectorTerms []v1.NodeSelectorTerm
-}
+type TopologyNodeFilter []v1alpha5.Requirements
 
-func NewTopologyNodeFilter(p *v1.Pod) *TopologyNodeFilter {
-	selector := &TopologyNodeFilter{
-		nodeSelector: p.Spec.NodeSelector,
-	}
-	if p.Spec.Affinity != nil && p.Spec.Affinity.NodeAffinity != nil && p.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
-		selector.nodeSelectorTerms = p.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms
+func MakeTopologyNodeFilter(p *v1.Pod) TopologyNodeFilter {
```

Comment:
```suggestion
func NewTopologyNodeFilter(p *v1.Pod) TopologyNodeFilter {
```

Reply:
`New.*` is typically reserved for methods that return a pointer interface while `Make.*` is for methods that return value types.  I'm good either way.

Reply:
I'm not familiar with this. Is this from the Go compiler?

Reply:
That's one place, it's not 100% followed consistently but it seems odd to have a New.*() method that doesn't return a pointer which is why I went with Make()

Go compiler
```
src/cmd/compile/internal/syntax/pos.go:24:func MakePos(base *PosBase, line, col uint) Pos { return Pos{base, sat32(line), sat32(col)} }
src/cmd/internal/src/pos.go:381:func makeLicoRaw(line, col uint) lico {
src/reflect/value.go:2912:func MakeSlice(typ Type, len, cap int) Value {
src/reflect/value.go:2931:func MakeChan(typ Type, buffer int) Value {
src/reflect/value.go:2947:func MakeMap(typ Type) Value {
src/reflect/value.go:2953:func MakeMapWithSize(typ Type, n int) Value {
src/reflect/value.go:3438:func makechan(typ *rtype, size int) (ch unsafe.Pointer)
```

Other projects use that idiom it as well:
```
vendor/go.starlark.net/starlark/int.go:43:func MakeInt64(x int64) Int {
vendor/go.starlark.net/starlark/int.go:51:func MakeUint(x uint) Int { return MakeUint64(uint64(x)) }
vendor/go.starlark.net/starlark/int.go:54:func MakeUint64(x uint64) Int {
vendor/go.starlark.net/starlark/int.go:67:func MakeBigInt(x *big.Int) Int {
```